src/common/health.rs

```diff
@@ -128,14 +128,20 @@ impl Task {
         while self.commit_index() < cluster_commit_index {
             // If not:
             //
-            // - Refresh cluster commit index
-            let Some(current_cluster_commit_index) = self.cluster_commit_index().await else {
+            // - Check if this is the only node in the cluster
+            let Some(_) = self.cluster_commit_index().await else {
                 self.set_ready();
                 return;
             };
 
-            // - Check if cluster commit index *decresed* since last check
-            cluster_commit_index = cmp::min(current_cluster_commit_index, cluster_commit_index);
+            // TODO: Do we want to update `cluster_commit_index` here?
+            //
+            // I.e.:
+            // - If we *don't* update `cluster_commit_index`, then we will only wait till the node
+            //   catch up with the cluster commit index *at the moment the node has been started*
+            // - If we *do* update `cluster_commit_index`, then we will keep track of cluster
+            //   commit index updates and wait till the node *completely* catch up with the leader,
+            //   which might be hard (if not impossible) in some situations
```

Comment:
First question: what do we want to do here?

Reply:
> catch up with the cluster commit index *at the moment the node has been started*

this should be fine, I guess.

There is a middle-ground of doing this update only once, after first sync is complete

Reply:
Ok, then it should be fine as-is.